include/irr/builtin/glsl/bxdf/common_samples.glsl

```diff
@@ -3,24 +3,37 @@
 
 #include <irr/builtin/glsl/bxdf/brdf/cos_weighted_sample.glsl>
 
-irr_glsl_BSDFSample irr_glsl_transmission_cos_sample(in irr_glsl_AnisotropicViewSurfaceInteraction interaction)
+irr_glsl_BSDFSample irr_glsl_transmission_cos_generate(in irr_glsl_AnisotropicViewSurfaceInteraction interaction)
 {
     irr_glsl_BSDFSample smpl;
     smpl.L = -interaction.isotropic.V.dir;
-    smpl.probability = 1.0;
+    //smpl.probability = 1.0;
     
     return smpl;
 }
 
-irr_glsl_BSDFSample irr_glsl_delta_cos_sample(in irr_glsl_AnisotropicViewSurfaceInteraction interaction)
+irr_glsl_transmission_cos_remainder_and_pdf(out float pdf, in irr_glsl_BSDFAnisotropicParams params, in irr_glsl_AnisotropicViewSurfaceInteraction interaction)
+{
+	pdf = 1.0;
+	return 0.0;
+}
+
+irr_glsl_BSDFSample irr_glsl_delta_cos_generate(in irr_glsl_AnisotropicViewSurfaceInteraction interaction)
 {
     irr_glsl_BSDFSample smpl;
     smpl.L = interaction.isotropic.N*2.0*interaction.isotropic.NdotV - interaction.isotropic.V.dir;
-    smpl.probability = 1.0;
+    //smpl.probability = 1.0;
 
     return smpl;
 }
 
+irr_glsl_delta_cos_remainder_and_pdf(out float pdf, in irr_glsl_BSDFAnisotropicParams params, in irr_glsl_AnisotropicViewSurfaceInteraction interaction)
```

Comment:
this is actually not technically a `delta` its a `reflection` (because delta means the function is 0 everywhere except for one point, reflection+transmission are both deltas, but the way the delta point is fixed is different)